android-vjestina-homework-movie-app-data-mock/app/src/main/java/agency/five/codebase/android/movieapp/ui/moviedetails/MovieDetailsScreen.kt

```diff
@@ -0,0 +1,283 @@
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.items
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.AsyncImage
+
+private val movieDetailsMapper: IMovieDetailsMapper = MovieDetailsMapperImpl()
+
+val movieDetailsViewState = movieDetailsMapper.toMovieDetailsViewState(MoviesMock.getMovieDetails())
+
+@Composable
+fun MovieDetailsRoute() {
+    //val movieDetailsViewState by remember { mutableStateOf(movieDetailsViewState) }
+    val movieDetailsViewState: MovieDetailsViewState = movieDetailsViewState
+
+    MovieDetailsScreen(
+        movieDetailsViewState = movieDetailsViewState
+    )
+}
+
+@Composable
+fun MovieDetailsScreen(
+    modifier: Modifier = Modifier,
+    movieDetailsViewState: MovieDetailsViewState
+) {
+    Column(
+        modifier = modifier
+            .verticalScroll(state = rememberScrollState())
+    ) {
+        MovieDetailsHeader(
+            movieDetailsViewState = movieDetailsViewState
+        )
+
+        Spacer(
+            modifier = Modifier
+                .height(MaterialTheme.spacing.medium)
+        )
+
+        MovieDetailsOverview(
+            movieDetailsViewState = movieDetailsViewState
+        )
+
+        Spacer(
+            modifier = Modifier
+                .height(MaterialTheme.spacing.medium)
+        )
+
+        MovieDetailsCrew(
+            movieDetailsViewState = movieDetailsViewState
+        )
+
+        Spacer(
+            modifier = Modifier
+                .height(MaterialTheme.spacing.medium)
+        )
+
+        MovieDetailsCast(
+            movieDetailsViewState = movieDetailsViewState
+        )
+    }
+}
+
+@Composable
+fun MovieDetailsHeader(
+    modifier: Modifier = Modifier
+        .fillMaxWidth(),
+    movieDetailsViewState: MovieDetailsViewState
+) {
+    Box(
+        modifier = modifier,
+    ) {
+        AsyncImage(
+            model = movieDetailsViewState.imageUrl,
+            contentDescription = movieDetailsViewState.title,
+            contentScale = ContentScale.Crop,
+            modifier = Modifier
+                .fillMaxSize()
```

Comment:
You should use `fillMaxWidth` here with a fixed `height` and align the image to the top. This might fix your issue with image loading.

Reply:
Did that but it doesn't work. Problem is that ActorCard pictures won't load on bottom of screen, it draws the name and role but not the image. Main image for movie worked before making suggested changes.